content/research/market-health/posts/2026-06-15-cross-pair-lot-fingerprints/analyze.py

```
@@ -0,0 +1,413 @@
+
+def archive_url(pair: str, date: str) -> str:
+    filename = f"{pair}-aggTrades-{date}.zip"
+    return f"{BASE_URL}/{pair}/{filename}"
+
+
+def download(pair: str, date: str, cache_dir: Path) -> tuple[Path, str]:
+    cache_dir.mkdir(parents=True, exist_ok=True)
+    path = cache_dir / f"{pair}-aggTrades-{date}.zip"
+    if not path.exists():
+        print(f"Downloading {archive_url(pair, date)}")
+        urllib.request.urlretrieve(archive_url(pair, date), path)
```

Ruff (0.15.15)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)